packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -451,11 +452,19 @@ class FlutterPlugin implements Plugin<Project> {
 
     // TODO(54566): Can remove this function and its call sites once resolved.
     /**
-     * Returns `true` if the given path contains an `android/build.gradle` file.
+     * Returns `true` if the given path contains an `android` directory
+     * containing a `build.gradle` or `build.gradle.kts` file.
      */
-    private static Boolean doesSupportAndroidPlatform(String path) {
-        File editableAndroidProject = new File(path, "android" + File.separator + "build.gradle")
-        return editableAndroidProject.exists()
+    private Boolean doesSupportAndroidPlatform(String path) {
+        File buildGradle = new File(path, 'android' + File.separator + 'build.gradle')
+        File buildGradleKts = new File(path, 'android' + File.separator + 'build.gradle.kts')
+        return buildGradle.exists() || buildGradleKts.exists()
+    }
+
+    private File settingsGradleFile() {
```

Comment:
Nit: Please make this helper function take in a project object instead of referencing the global project variable. It will help when we start pulling out code from flutter.groovy. Plus there are some difficult corner cases for when it is ok to reference the project object, for example in gradle 8+ project is null in a doAfter block, and it will make it more obvious when we are depending on the project object.

Reply:
Done.

> in gradle 8+ project is null in a doAfter block

Thanks, good to know! I can imagine this knowledge is hard earned, haha.

Reply:
Yeah 2+ weeks right before christmas where this is all I did lol.